recipes/qt/5.x.x/conanfile.py

```diff
@@ -272,6 +240,48 @@ def _enablemodule(mod):
             if self.options.get_safe(module):
                 _enablemodule(module)
 
+    def validate(self):
+        if self.options.widgets and not self.options.gui:
+            raise ConanInvalidConfiguration("using option qt:widgets without option qt:gui is not possible. "
+                                            "You can either disable qt:widgets or enable qt:gui")
+
+        if self.options.qtwebengine:
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("Static builds of Qt Webengine are not supported")
+
+            if tools.cross_building(self.settings, skip_x64_x86=True):
+                raise ConanInvalidConfiguration("Cross compiling Qt WebEngine is not supported")
+
+            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+            if self.settings.os == "Linux":
+                #Linking of QtWebEngine shared library fails with unresolved error to freetype.
+                raise ConanInvalidConfiguration("Building QtWebEngine on Linux will currently result in a build error.")
```

Comment:
Does the error match a bugreport upstream ?

Reply:
I haven't found one yet. It is possible that this issue only happens on Suse systems. I built the bincrafters Qt5 recipe and got the same link error. After office hours I can build the recipe on an Ubuntu system to see if it succeeds there.

Reply:
Building qt5 package with qt:qtwebengine=True on Ubuntu 20 fails already during compilation stage, see below. Debugging this is complicated because the chromium build system is driven by qmake, so multiple levels. Also I am running out of time that I can spend on this, so I will leave QtWebEngine on Linux as an invalid configuration.
```
Project MESSAGE: Running: /home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/3rdparty/gn/out/Release/gn gen /home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/pdf/release --script-executable=/usr/bin/python2 --args='use_qt=true init_stack_vars=false is_component_build=false is_shared=true enable_debugallocation=false enable_media_remoting=false enable_message_center=false enable_nacl=false enable_remoting=false enable_reporting=false enable_resource_whitelist_generation=false enable_swiftshader=false angle_enable_swiftshader=false enable_web_auth=true enable_web_speech=false enable_widevine=true forbid_non_component_debug_builds=false has_native_accessibility=false safe_browsing_mode=0 toolkit_views=false treat_warnings_as_errors=false use_allocator_shim=false use_allocator="none" use_custom_libcxx=false closure_compile=false optimize_webui=false v8_use_external_startup_data=false strip_absolute_paths_from_debug_symbols=false use_jumbo_build=true jumbo_file_merge_limit=8 jumbo_build_excluded=["browser"] enable_precompiled_headers=true is_official_build=true is_debug=false symbol_level=0 blink_symbol_level=0 remove_v8base_debug_symbols=true use_cups=false use_gio=false use_gnome_keyring=false linux_use_bundled_binutils=false use_udev=true use_bundled_fontconfig=false use_sysroot=false enable_session_service=false is_cfi=false use_ozone=true ozone_auto_platforms=false ozone_platform_headless=false ozone_platform_external=true ozone_platform="qt" ozone_extra_path="/home/robocoop/.conan/data/qt/5.15.2/_/_/source/qt5/qtwebengine/src/core/ozone/ozone_extra.gni" use_gold=false use_lld=false is_clang=false custom_toolchain="/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/toolchain:target" host_toolchain="/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/toolchain:host" host_cpu="x64" pkg_config="pkg-config" host_pkg_config="/usr/bin/pkg-config" pdfium_use_system_zlib=true use_system_libpng=true pdfium_use_system_libpng=true use_system_libjpeg=false use_system_freetype=true use_system_harfbuzz=false use_glib=false pdf_enable_v8=false pdf_enable_xfa=false pdf_enable_xfa_bmp=false pdf_enable_xfa_gif=false pdf_enable_xfa_png=false pdf_enable_xfa_tiff=false qtwebengine_target="/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/pdf/release:QtPdf"' --root=/home/robocoop/.conan/data/qt/5.15.2/_/_/source/qt5/qtwebengine/src/3rdparty/chromium
ERROR at //build/config/linux/pkg_config.gni:103:17: Script returned non-zero exit code.
    pkgresult = exec_script(pkg_config_script, args, "value")
                ^----------
Current dir: /home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/pdf/release/
Command: /usr/bin/python2 /home/robocoop/.conan/data/qt/5.15.2/_/_/source/qt5/qtwebengine/src/3rdparty/chromium/build/config/linux/pkg-config.py -p pkg-config nss -v -lssl3
Returned 1.
stderr:

Package nss was not found in the pkg-config search path.
Perhaps you should add the directory containing `nss.pc'
to the PKG_CONFIG_PATH environment variable
No package 'nss' found
Could not run pkg-config.

See //build/config/linux/nss/BUILD.gn:15:3: whence it was called.
  pkg_config("system_nss_no_ssl_config") {
  ^---------------------------------------
See //crypto/BUILD.gn:214:25: which caused the file to be included.
    public_configs += [ "//build/config/linux/nss:system_nss_no_ssl_config" ]
                        ^--------------------------------------------------
Project ERROR: GN run error!
make[3]: *** [Makefile:65: sub-gn_run-pro-make_first] Error 3
make[3]: Leaving directory '/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src/pdf'
make[2]: *** [Makefile:101: sub-pdf-make_first] Error 2
make[2]: Leaving directory '/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine/src'
make[1]: *** [Makefile:49: sub-src-make_first] Error 2
make[1]: Leaving directory '/home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955/build_folder/qtwebengine'
make: *** [Makefile:99: module-qtwebengine-make_first] Error 2
qt/5.15.2:
qt/5.15.2: ERROR: Package '9561b33a597e07c59d1245673bc56880e181f955' build failed
qt/5.15.2: WARN: Build folder /home/robocoop/.conan/data/qt/5.15.2/_/_/build/9561b33a597e07c59d1245673bc56880e181f955
ERROR: qt/5.15.2: Error in build() method, line 653
        self.run(self._make_program(), run_environment=True)
        ConanException: Error 2 while executing make
```